### PR TITLE
Fixing some issues with documentation

### PR DESCRIPTION
**Pull request by wetted**

### CODE REVIEW

File: src/main/docs/guide/quickStart/bsonQuick.adoc [hunk 1]
```diff
@@ -7,13 +7,17 @@ To do so in a Gradle build add the following configuration (either Groovy or Kot
 configurations.all {
     resolutionStrategy.dependencySubstitution {
         substitute(module("io.micronaut:micronaut-jackson-databind"))
-            .using(module("io.micronaut.serde:micronaut-serde-bson:{version}"))
+            .using(module("jakarta.json.bind:jakarta.json.bind-api:2.0.0"))
```
Comment (graemerocher) [inline src/main/docs/guide/quickStart/bsonQuick.adoc:10]: I don't understand this, this section relates to BSON?

Comment (wetted) [reply src/main/docs/guide/quickStart/bsonQuick.adoc:10]: @graemerocher Yes, this section is for BSON. The current `micronaut-serialization` docs have:

```
configurations.all {
    resolutionStrategy.dependencySubstitution {
        substitute(module("io.micronaut:micronaut-jackson-databind"))
            .using(module("io.micronaut.serde:micronaut-serde-bson:1.0.1"))
        substitute(module("io.micronaut:micronaut-jackson-core"))
            .using(module("io.micronaut.serde:micronaut-serde-bson:1.0.1"))
    }
}
```

If I create a project with the `Book` example of the docs, it doesn't compile. 

However, the JSONP/JSONB section has 

```
configurations.all {
    resolutionStrategy.dependencySubstitution {
        substitute(module("io.micronaut:micronaut-jackson-databind"))
            .using(module("jakarta.json.bind:jakarta.json.bind-api:2.0.0"))
        substitute(module("io.micronaut:micronaut-jackson-core"))
            .using(module("io.micronaut.serde:micronaut-serde-jsonp:1.0.1"))
    }
}
```

If I make the BSON example consistent with the JSONP/JSONB by using the `jakarta.json.bind:jakarta.json.bind-api` substitution, then it compiles and passes `BookTest`.

Have I misunderstood? If so, I also need to address it in the starter feature as well, on [PR 1189](https://github.com/micronaut-projects/micronaut-starter/pull/1189)

Comment (yawkat) [reply src/main/docs/guide/quickStart/bsonQuick.adoc:10]: @wetted Which book example are you using? I looked at the guide at https://micronaut-projects.github.io/micronaut-serialization/latest/guide/#bsonQuick and it has this example, which should work without jakarta.json.bind-api:

```java
package example;

import io.micronaut.serde.annotation.Serdeable;
import org.bson.codecs.pojo.annotations.BsonCreator;
import org.bson.codecs.pojo.annotations.BsonProperty;

@Serdeable // 
public class Book {
    private final String title;
    @BsonProperty("qty") // 
    private final int quantity;

    @BsonCreator // 
    public Book(String title, int quantity) {
        this.title = title;
        this.quantity = quantity;
    }

    public String getTitle() {
        return title;
    }

    public int getQuantity() {
        return quantity;
    }
}
```

If there's documentation somewhere that requires jakarta.json.bind-api for the bson stack, those docs are wrong. jakarta.json.bind-api and bson should not usually be present in the same project

Comment (wetted) [reply src/main/docs/guide/quickStart/bsonQuick.adoc]: Nevermind. I was mistaken and have corrected with a subsequent commit.

**Pull request merged**